src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -588,14 +589,17 @@ def _prepare_attn_mask(self, attention_mask, input_shape, inputs_embeds, past_ke
         # create causal mask
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
         combined_attention_mask = None
+        device = attention_mask.device
+        dtype = torch.bool
+
         if input_shape[-1] > 1:
             combined_attention_mask = _make_causal_mask(
-                input_shape, inputs_embeds.dtype, past_key_values_length=past_key_values_length
-            ).to(attention_mask.device)
+                input_shape, dtype, device, past_key_values_length=past_key_values_length
+            )
 
         if attention_mask is not None:
             # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-            expanded_attn_mask = _expand_mask(attention_mask, inputs_embeds.dtype, tgt_len=input_shape[-1])
+            expanded_attn_mask = _expand_mask(attention_mask, dtype, tgt_len=input_shape[-1])
```

Comment:
I can't comment on line 604, but since everything is bool now, the + should be changed to | to make it clear to the reader this is a logical or and not an addition.